### ISSUE TITLE
[Package]: PHP 7

### ISSUE THREAD

**Issue by fancsali** [termux/termux-packages#7446]
### Package description

May I suggest to include a package named php7 alongside the most up-to-date php8. Perhaps it could be implemented in a similar manner as in the other APT-based distributions: Have two packages php7 and php8 and then php is just a virtual one pointing to the latest one.

Although this seems unnecessary at first sight, as there are several projects _trying_ to solve this issue, like:

- https://github.com/take-i/termux-php7
- https://github.com/Termux-pod/termux-pod

However, every those would routinely have issues with library versions, and since Termux does not have a concept of installing old (or specific) versions of (dependent) packages, it's a pain to maintain such a hybrid system.

### Home page URL

https://www.php.net/

### Source code URL

https://www.php.net/downloads.php

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

**Comment by xtkoba** [comment]
I suppose this is not unlikely to happen. For scripting languages, we already have examples of Guile, Lua and Python.

**Comment by fancsali** [comment]
Exactly my point: most of the scripting languages in most of the *nix distributions seem to have a multiple packages (see php7 and php8 in Debian as an example), and then the meta package pointing to the latest. (In the example case php -> php8).

And, I can see the same logic being applied in Termux as well; as mentioned before Lua is a good example...

**Comment by thunder-coding** [comment]
Well, I want to say that unlike nodejs which even takes 3+ hours to build on CI, building php hardly takes half an hour. So, wouldn't it make sense to build it yourself with patches borrowed from here. Also, we are limited in terms of disk space on our official apt mirrors

**Comment by fancsali** [comment]
Well, I am not sure, that's the case on a (relatively) underpowered phone or tablet.

And then again, apologies, but I fail to see the difference between Lua and PHP in this regard.

The disk space issue is the real issue here, I reckon.

**Comment by Grimler91** [comment]
We can add it I guess if many people want it, but would be nice if someone who actually uses the package would volunteer and maintain it (i.e. keep the package up to date with latest php7.X.X release, and test it)

**Comment by eeepyn** [comment]
I really want to php7.x.x，because php8 is not matched with my thinkphp 6.

**Comment by xtkoba** [comment]
Yup, at least it builds: <a href="https://github.com/termux/termux-packages/actions/runs/1621754358">PHP 7.4.12</a>

**Comment by xtkoba** [comment]
<a href="https://github.com/termux/termux-packages/actions/runs/1631561621">PHP 7.4.27</a>

**Comment by mustaphatg** [comment]
Thanks so much for the great job. Am from Nigeria and you really Helped me. I can't code on my system for long as the power supply is not stable like in the US.
so I use termux most times but not until I got a new android phone and unfortunately it was an android 11. On installing PHP, I got PHP 8.x.x and you know its hard to come by a Laravel 8 project. Most freelance jobs on Fiverr are based on laravel 5,6 . And with php 8, all you'll be left with is a depreciation error of incompatible return types of ArrayAccess. In short thanks so much,, I now have PHP 7.4.x on my android machine.
Thanks so much
